packages/scroll-sync/src/ScrollSyncPane.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+	Children,
+	cloneElement,
+	forwardRef,
+	isValidElement,
+	type ReactElement,
+	type RefAttributes,
+	useEffect,
+	useRef,
+	FunctionComponentElement,
+	ForwardedRef,
+} from 'react';
+import { useScrollSync } from './ScrollSyncContext';
+import { useMergeRefs } from './useMergeRefs';
+
+export interface IScrollSyncPaneProps {
+	children: ReactElement<RefAttributes<unknown>>;
+}
+
+export const ScrollSyncPane = forwardRef<unknown, IScrollSyncPaneProps>(function ScrollSyncPaneRef({ children }, ref) {
+	const childRef = useRef<HTMLElement | null>(null);
+	const { registerPane, unregisterPane } = useScrollSync();
+
+	useEffect(() => {
```

Comment:
Is it better for this to be useLayoutEffect?
Is it important to attach listeners before browser paint in case auto scrollTo on browser back kicks in?